shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -1021,6 +1072,18 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool, val ne
       case pat if ctx.inPattern =>
         err(msg"Unsupported pattern shape${
           if (dbg) " ("+pat.getClass.toString+")" else ""}:", pat.toLoc)(raise)
+      case Lam(pat, body) if ctx.inQuote =>
+        println(s"TYPING QUOTED LAM")
+        ctx.nest.copy(qenv = MutMap.empty, fvars = MutSet.empty).poly { newCtx =>
```

Comment:
Why do you do `.copy(qenv = MutMap.empty, fvars = MutSet.empty)` here instead of making `nest` correct by construction by having it do it? There might be calls to `nest` that you forget to update.

Reply:
I added a new `qnest` function for this situation.